[KB-204] Stacked Generalization (#166)
chhoumann/thesis-chemcam: report_thesis/src/sections/background.tex

\subsubsection{XGBoost}

\subsubsection{Stacked Generalization}

> Stacked generalization, introduced by \citet{wolpertstacked_1992}, is a technique designed to improve the predictive performance of machine learning models by leveraging the strengths of multiple models.

Reviewer: ```suggestion
Stacked generalization, introduced by \citet{wolpertstacked_1992}, is a method designed to improve the predictive performance of machine learning models by leveraging the strengths of multiple models.
```

> In stacked generalization, multiple base models (level-0 generalizers) are trained on the original dataset. 

Reviewer: ```suggestion
In this technique, multiple predictors are trained on the original dataset. 
```

> The predictions of these base models are then used as inputs to a meta-model (level-1 generalizer), which is trained to make the final prediction. 

Reviewer: ```suggestion
The outputs of these base models serve as inputs to a meta-model, which is trained to make the final prediction.
```

> This approach allows the meta-model to learn how to best combine the outputs of the base models to minimize generalization error.

Reviewer: ```suggestion
This strategy enables the meta-model to learn the optimal way to combine the outputs of the base models to minimize the generalization error.
```

> To formalize, let $\mathbf{X}$ be the input data and $\mathbf{y}$ be the target variable. 
> First, $N$ base models $G_1, G_2, \ldots, G_N$ are trained on the dataset $\mathbf{X}$. 
> Each base model produces a set of predictions $\hat{\mathbf{y}}_i = G_i(\mathbf{X})$.

Reviewer: ```suggestion
Formally, let $\mathbf{X}$ denote the input data and $\mathbf{y}$ the target variable.
Initially, $N$ base models $G_1, G_2, \ldots, G_N$ are trained on the dataset $\mathbf{X}$.
Each base model generates a set of predictions $\hat{\mathbf{y}}_i = G_i(\mathbf{X})$.
```

> The predictions from the base models are then combined into a new dataset $\mathbf{Z}$, where each column $\mathbf{z}_i \in \mathbf{Z}$ corresponds to the predictions from the $i$-th base model:

Reviewer: ```suggestion
The predictions from the base models are then compiled into a new dataset $\mathbf{Z}$, where each column $\mathbf{z}_i \in \mathbf{Z}$ represents the predictions from the $i$-th base model:
```

$$
\mathbf{Z} = [\hat{\mathbf{y}}_1, \hat{\mathbf{y}}_2, \ldots, \hat{\mathbf{y}}_N]
$$

> A meta-model $F$ is then trained on this new dataset $\mathbf{Z}$ to predict the target variable $\mathbf{y}$:

Reviewer: ```suggestion
A meta-model $F$ is subsequently trained on this new dataset $\mathbf{Z}$ to predict the target variable $\mathbf{y}$:
```

$$
\mathbf{\hat{y}} = F(\mathbf{Z})
$$

> The final prediction is given by the meta-model $F$, which effectively combines the outputs of the base models in an optimal way to improve overall performance. 

Reviewer: ```suggestion
The final prediction is provided by the meta-model $F$, which effectively integrates the outputs of the base models to enhance overall performance.
```

> The power of stacked generalization lies in its ability to exploit the unique strengths of different base models and mitigate their individual weaknesses, leading to a more accurate and generalizable ensemble model.

Reviewer: ```suggestion
The efficacy of stacked generalization lies in its capability to leverage the distinct strengths of various base models while compensating for their individual weaknesses, resulting in a more accurate and generalizable ensemble model.
```

In this technique, multiple base models are trained on the original dataset. 
The outputs of these base models serve as inputs to a meta-model, which is trained to make the final prediction.
This strategy enables the meta-model to learn the optimal way to combine the outputs of the base models to minimize the generalization error.

Formally, let $\mathbf{X}$ denote the input data and $\mathbf{y}$ the target variable.
Initially, $N$ base models $G_1, G_2, \ldots, G_N$ are trained on the dataset $\mathbf{X}$.
Each base model generates a set of predictions $\hat{\mathbf{y}}_i = G_i(\mathbf{X})$.

The predictions from the base models are then compiled into a new dataset $\mathbf{Z}$, where each column $\mathbf{z}_i \in \mathbf{Z}$ represents the predictions from the $i$-th base model:

$$
\mathbf{Z} = [\hat{\mathbf{y}}_1, \hat{\mathbf{y}}_2, \ldots, \hat{\mathbf{y}}_N]
$$

A meta-model $F$ is subsequently trained on this new dataset $\mathbf{Z}$ to predict the target variable $\mathbf{y}$:

$$
\mathbf{\hat{y}} = F(\mathbf{Z})
$$

The final prediction is provided by the meta-model $F$, which effectively integrates the outputs of the base models to enhance overall performance.

> 
> The efficacy of stacked generalization lies in its capability to leverage the distinct strengths of various base models while compensating for their individual weaknesses, resulting in a more accurate and generalizable ensemble model.

Reviewer 2: ```suggestion
The effectiveness of stacked generalization stems from its ability to leverage the unique strengths of different base models while mitigating their individual weaknesses, thereby producing a more accurate and generalizable ensemble model.
```
